x-pack/plugins/apm/server/lib/transactions/charts/get_anomaly_data/index.ts

```diff
@@ -75,10 +83,6 @@ export async function getAnomalySeries({
     return;
   }
 
-  // don't fetch anomalies if there are isn't exaclty 1 ML job match for the given environment
-  if (mlJobIds.length !== 1) {
-    return;
-  }
```

Comment:
Shouldn't you keep this check?

Reply:
Afaict this should only happen if the user has selected `ALL` environments. I've added an explicit check for that above.